midend/lib/Conversion/MatMulOptimization/BatchMatMulOptimize.cpp

```diff
@@ -81,16 +81,28 @@ class BatchMatMulOptimizePattern : public ConversionPattern {
 
     const Value zeroElementType = rewriter.create<arith::ConstantOp>(
         loc, rewriter.getZeroAttr(elementType));
-    const Value zeroElementTypeVec = rewriter.create<vector::SplatOp>(
-        loc, VectorType::get({affineVectorSize}, elementType), zeroElementType);
+
+    const Value zeroElementTypeVec =
+        isa<IntegerType>(elementType)
+            ? rewriter
+                  .create<vector::BroadcastOp>(
+                      loc, VectorType::get({affineVectorSize}, elementType),
+                      zeroElementType)
+                  .getResult()
+            : rewriter
+                  .create<vector::SplatOp>(
+                      loc, VectorType::get({affineVectorSize}, elementType),
+                      zeroElementType)
+                  .getResult();
```

Comment:
What is the reason for this change(If there are actual changes, there is no need to add comments in this code, I just wonder)?